### PR TITLE
Adding mongo headless service to mongo sts

### PR DESCRIPTION
**Pull request by rajdas98**
Signed-off-by: rajdas98 <mail.rajdas@gmail.com>

## Proposed changes

* adding mongo headless svc 
* Replacing mongo-sts serviceName with mongo-headless

Refer- https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#limitations
## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: litmus-portal/namespaced-k8s-template.yml [hunk 1]
```diff
@@ -697,3 +697,18 @@ spec:
...
   selector:
     component: database
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: mongo
+  name: mongo-headless-service
+  namespace: litmus
```
Comment (Jonsy13) [inline litmus-portal/namespaced-k8s-template.yml:707]: Namespace not required for namespaced-mode

File: litmus-portal/cluster-k8s-manifest.yml [hunk 2]
```diff
@@ -727,3 +727,18 @@ spec:
       targetPort: 27017
   selector:
     component: database
+---
```
Comment (Jonsy13) [inline litmus-portal/cluster-k8s-manifest.yml:730]: Do we need above service then?

File: litmus-portal/namespaced-k8s-template.yml [hunk 3]
```diff
@@ -697,3 +697,18 @@ spec:
       targetPort: 27017
   selector:
     component: database
+---
```
Comment (Jonsy13) [inline litmus-portal/namespaced-k8s-template.yml:700]: Same here, Do we need above service then?

**Review by Jonsy13 — CHANGES_REQUESTED** [review]

File: litmus-portal/cluster-k8s-manifest.yml (hunk 2 above)
Comment (rajdas98) [reply litmus-portal/cluster-k8s-manifest.yml:730]: yes, headless service is not accessible outside. It's different than the normal service

File: litmus-portal/namespaced-k8s-template.yml (hunk 1 above)
Comment (rajdas98) [reply litmus-portal/namespaced-k8s-template.yml]: done

**Comment by gdsoumya** [comment]
Why is this needed? Would be good if you can add more context to the PR or create a related issue.

**Comment by rajdas98** [comment]
As mentioned for best practices to have a headless service in STS. Refer- https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#limitations

**Pull request merged**